opm/autodiff/StandardWell_impl.hpp

```diff
@@ -35,7 +35,7 @@ namespace Opm
     , perf_pressure_diffs_(number_of_perforations_)
     , primary_variables_(numWellEq, 0.0)
     , primary_variables_evaluation_(numWellEq) // the number of the primary variables
-    , F0_(numWellEq)
+    , F0_(num_components_)
     {
```

Comment:
Why did you change this? It should be number of equations.

Reply:
Nope. It only applies to mass balance equations, not well control equation. It should be `numWellEq - 1`  or `num_components_`.

Reply:
Ok. I get it. numWellEq == num_components + num_control_eq. Then I agree on your change.

Reply:
So now we have for 
: blackoil 
num_components = 3 numWelleq = 4
: polymer
num_components = 3 numWellEq = 4 
: solvent
num_componets = 4 numWellEq = 4

Maybe it is better with an extra variable` numWellConservationEq` 
Then you get: 
```
num_components 
numWellConservationEq (== num_components for all but the polymer case) 
numWellEq (== numWellConservationEq + 1)
```

The current implementation (also before this PR) are too often mixing the concept of number of components, phases and equations.

Reply:
> : solvent
num_componets = 4 numWellEq = 4

So we do not have a mass balance equation for the solvent model?  The solvent part is something difficult for me, since I do not understand the details.


> The current implementation (also before this PR) are too often mixing the concept of number of components, phases and equations.

It has been always been this way.  We should make clear that what is the equation index, what is the primary variable index. (component/phase thing is a little harder and I do not have good suggestions, but we should provide a fix/convention eventually).